pyriemann/spatialfilters.py

```diff
@@ -467,3 +469,266 @@ def fit(self, X, y):
+    overlap : float (default 0.5)
+        The percentage of overlap between window.
+    fmin : float | None , (default None)
+        The minimal frequency to be returned. Since BSS models assume zero-mean
+        processes, the first cospectrum (0 Hz) must be excluded.
+    fmax : float | None , (default None)
+        The maximal frequency to be returned.
+    fs : float | None , (default None)
+        The sampling frequency of the signal.
+    expl_var : float (default 0.999)
+        The percentage of explained variance in ]0, 1], for dimension reduction
+        of cospectra, because Pham's AJD is sensitive to matrices conditioning.
+    verbose : bool (default True)
+        Verbose flag.
+
+    Attributes
+    ----------
+    n_channels_ : int
+        If fit, the number of channels of the signal.
+    freqs_ : ndarray, shape (n_freqs,)
+        If fit, the frequencies associated to cospectra.
+    n_sources_ : int
+        If fit, the number of components of the source space.
+    forward_filters_ : ndarray, shape ``(n_sources_, n_channels_)``
+        If fit, the spatial filters used to transform signal into source,
+        also called deximing or separating matrix.
+    backward_filters_ : ndarray, shape ``(n_channels_, n_sources_)``
+        If fit, the spatial filters used to transform source into signal,
+        also called mixing matrix.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.7.dev
+
+    See Also
+    --------
+    CospCovariances
+
+    References
+    ----------
+    [1] M. Congedo, C. Gouy-Pailler, C. Jutten, "On the blind source separation
+    of human electroencephalogram by approximate joint diagonalization of
+    second order statistics", Clin Neurophysiol, 2008
+
+    [2] M. Congedo, R. John, D. De Ridder, L. Prichep, "Group indepedent
+    component analysis of resting state EEG in large normative samples",
+    Int J Psychophysiol, 2010
+
+    [3] D.-T. Pham, "Joint approximate diagonalization of positive definite
+    Hermitian matrices", SIAM J Matrix Anal Appl, 2001
+    """
+
+    def __init__(self, window=128, overlap=0.5, fmin=None, fmax=None, fs=None,
+                 expl_var=0.999, verbose=True):
+        """Init."""
+        if not 0 < expl_var <= 1:
+            raise ValueError('Parameter expl_var must be included in ]0, 1]')
+
+        self.window = window
+        self.overlap = overlap
+        self.fmin = fmin
+        self.fmax = fmax
+        self.fs = fs
+        self.expl_var = expl_var
+        self.verbose = verbose
+
+    def fit(self, X, y=None):
+        """Fit.
+
+        Compute and diagonalize cospectra, to estimate forward and backward
+        spatial filters.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_subjects, n_conditions, n_channels, n_samples) | list of n_subjects of list of n_conditions ndarray of shape (n_channels, n_samples), with same n_conditions and n_channels but different n_samples
```

Comment:
not sure what is the convention in pyRiemann but I would use n_times and not n_samples here. n_samples is confusing with sklearn naming

Reply:
In my PR, I have simply re-used the dimension names of the Covariances class: https://github.com/pyRiemann/pyRiemann/blob/master/pyriemann/estimation.py#L67

But I do agree that _samples_ is an ambiguous term, with different interpretations depending the context. What about another issue to homogenize dimension names in pyRiemann (partially related to #25)?